src/lib/ngsiNotify/QueueNotifier.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+
+#include "common/string.h"
+
+#include "ngsiNotify/QueueNotifier.h"
+
+/* ****************************************************************************
+*
+* Notifier::sendNotifyContextRequest -
+*/
+void QueueNotifier::sendNotifyContextRequest(NotifyContextRequest* ncr, const std::string& url, const std::string& tenant, const std::string& xauthToken, Format format)
+{
+  ConnectionInfo ci;
```

Comment:
A big chunk of the code in this method is duplicated in the basic Notifier, e.g. spathList generation. Probably it doesn't make sense to make this PR more complex refactoring common code in external functions, but at least a FIXME could be added in the preamble of the function, e.g.:

"FIXME P5: analyze how much of the code of this function is the same than in Notifier::sendNotifyContextRequest and could be refactored to common functions"